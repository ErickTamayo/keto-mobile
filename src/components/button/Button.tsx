import * as React from 'react'
import { TouchableOpacity, Text, View, GestureResponderEvent } from 'react-native'
import { buttonStyles, buttonContainerStyle, disabledStyle } from './styles'

export interface Props {
  title: string
  variant: 'primary' | 'primary-md' | 'primary-lg' | 'secondary' | 'secondary-lg'
  onPress?: (event: GestureResponderEvent) => void
  disabled?: boolean
  loading?: boolean
}

const Button = ({ title, variant, onPress, disabled }: Props) => (
  <TouchableOpacity
    style={[...buttonStyles[variant].button, disabled ? disabledStyle : null]}
    onPress={onPress}
    disabled={disabled}
  >
    <View style={buttonContainerStyle}>
      <Text style={buttonStyles[variant].text}>{title}</Text>
    </View>
  </TouchableOpacity>
)

export default Button
