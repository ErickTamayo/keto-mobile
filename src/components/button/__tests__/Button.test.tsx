import 'react-native'
import * as React from 'react'
import { TouchableOpacity, GestureResponderEvent } from 'react-native'
import { shallow } from 'enzyme'
import Button, { Props } from '../Button'

describe('<Button />', () => {
  it('renders correctly', () => {
    const props: Props = {
      variant: 'primary',
      title: 'button-title',
    }
    const wrapper = shallow(<Button {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders disabled', () => {
    const props: Props = {
      variant: 'primary',
      title: 'button-title',
      disabled: true,
    }
    const wrapper = shallow(<Button {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should trigger onPress once pressed', () => {
    const props: Props = {
      variant: 'primary',
      title: 'button-title',
      onPress: jest.fn(),
    }
    const wrapper = shallow(<Button {...props} />)
    wrapper.find(TouchableOpacity).prop('onPress')!({} as GestureResponderEvent)

    expect(props.onPress).toBeCalled()
  })
})
