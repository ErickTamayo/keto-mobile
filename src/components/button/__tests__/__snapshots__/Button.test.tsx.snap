// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Array [
      Object {
        "margin": 6.5,
      },
      Object {
        "paddingLeft": 19.5,
      },
      Object {
        "paddingRight": 19.5,
      },
      Object {
        "height": 48,
      },
      Object {
        "borderWidth": 1,
      },
      Object {
        "borderColor": "#57ae5b",
      },
      Object {
        "borderRadius": 100,
      },
      Object {
        "backgroundColor": "#57ae5b",
      },
      null,
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "fontFamily": "avenir-next-medium",
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "fontSize": 16,
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      button-title
    </Component>
  </Component>
</TouchableOpacity>
`;

exports[`<Button /> renders disabled 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  disabled={true}
  style={
    Array [
      Object {
        "margin": 6.5,
      },
      Object {
        "paddingLeft": 19.5,
      },
      Object {
        "paddingRight": 19.5,
      },
      Object {
        "height": 48,
      },
      Object {
        "borderWidth": 1,
      },
      Object {
        "borderColor": "#57ae5b",
      },
      Object {
        "borderRadius": 100,
      },
      Object {
        "backgroundColor": "#57ae5b",
      },
      Array [
        Object {
          "backgroundColor": "#cfcfcf",
        },
        Object {
          "borderColor": "transparent",
        },
      ],
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "fontFamily": "avenir-next-medium",
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "fontSize": 16,
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      button-title
    </Component>
  </Component>
</TouchableOpacity>
`;
